src/components/chart/specific-chart-options.js

```diff
@@ -38,6 +81,80 @@ class SpecificChartOptions {
             },
         };
     };
+
+    updateLegendSymbols = (chart) => {
+        if (chart.legend.options.enabled) {
+            chart.legend.allItems.forEach((item, index) => {
+                const { legendItem, userOptions } = item;
+                const seriesType = userOptions?.type;
+                const { label, symbol, line } = legendItem || {};
+
+                if (seriesType === 'line') {
+                    // This is the case for the column plus line chart - the series type is
+                    // line, but the chart type is column. In this case we show a simple
+                    // line symbol in the legend, but we need to move the label to the right
+                    // to account for the longer line symbol
+                    if (chart.userOptions.chart.type !== 'line') {
+                        label?.attr({
+                            x: 30, // Adjust label position to account for longer line
+                        });
+                    }
+
+                    // This is the scenario for a line chart with markers disabled
+                    // We have custom code in line-chart.js to update the last point to
+                    // display as a symbol. This code checks if there is no symbol in the legend
+                    // (which means it is a line chart with markers disabled)
+                    // and if so, it updates the legend to display as a symbol rather than as a line
+                    // We only to this for chart types that are explicitly line charts - i.e. not column with line
+                    if (!symbol && label && label.element && chart.userOptions.chart.type === 'line') {
+                        // Hide the line in the legend
+                        if (line) {
+                            line.hide();
+                        }
+
+                        // Remove any existing custom symbols for this legend item
+                        const existingSymbols = label.parentGroup.element.querySelectorAll('[data-custom-legend-symbol]');
```

Comment:
Additional code here to remove any existing symbols (I found when I moved this function to be called in the render event that the custom symbols were duplicated)